File: go/internal/answer/p6/p6.go
package p6

import "fmt"

type Sequence struct {
	n uint64
}

func (s *Sequence) sumOfSquares() uint64 {
	return s.n * (s.n + 1) * (2*s.n + 1) / 6
}

func (s *Sequence) sum() uint64 {
	return (1 + s.n) * s.n / 2
}

func Example() {
	s := Sequence{n: 100}
	sum := s.sum()
	diff := sum*sum - s.sumOfSquares()
	fmt.Println(diff)
	// Output: 25164150
}

File: typescript/p6/p6.ts
export function assert(condition: any, msg?: string): asserts condition {
  if (!condition) {
    throw new Error(msg);
  }
}

interface Sequence {
  n: number;
}

function sumOfSquares(s: Sequence): number {
  return s.n * (s.n + 1) * (2 * s.n + 1) / 6;
}

function sum(s: Sequence): number {
  return (1 + s.n) * s.n / 2;
}

const s: Sequence = { n: 100 };
const su = sum(s);
const diff = su * su - sumOfSquares(s);
console.log(diff);
assert(diff === 25164150);
